Lib/fontParts/base/groups.py

```diff
@@ -160,186 +236,269 @@ def _get_side2KerningGroups(self):
+            is not found`. The value must be either a class`list` or :class:`tuple`
+            of :class:`str` glyph names, or :obj:`None`. Defaults to :obj:`None`.
+        :return: The contents of the given group as a :class:`tuple`
+            of :class:`str` items, or the `default` value if the group is not found.
+
+        Example::
 
             >>> font.groups["myGroup"]
             ("A", "B", "C")
 
-        It is important to understand that any changes to the returned group
-        contents will not be reflected in the Groups object. If one wants to
-        make a change to the group contents, one should do the following::
+        ..note::
+
+            Any changes to the returned lib contents will not be reflected in
+            it's :class:`BaseGroups` instance. To make changes to this content,
+            do the following::
+
+                >>> group = font.groups["myGroup"]
+                >>> group.remove("A")
+                >>> font.groups["myGroup"] = group
 
-            >>> group = font.groups["myGroup"]
-            >>> group.remove("A")
-            >>> font.groups["myGroup"] = group
         """
         return super(BaseGroups, self).get(groupName, default)
 
-    def items(self):
-        """
-        Returns a list of ``tuple`` of each group name and group members.
-        Group names are :ref:`type-string` and group members are a
-        :ref:`type-immutable-list` of :ref:`type-string`. The initial
-        list will be unordered.
+    def items(self) -> BaseItems[Tuple[str, ValueType]]:
+        """View the items in the current groups instance.
+
+        Each item is represented as a :class:`tuple` of key-value pairs, where:
+            - `key` is a :class:`str` representing a group name.
+            - `value` is a :class:`tuple` of :class:`str` glyph names.
+
+        :return: A :ref:`type-view` of the groups' ``(key, value)`` pairs.
+
+        Example::
 
             >>> font.groups.items()
-            [("myGroup", ("A", "B", "C"), ("myGroup2", ("D", "E", "F"))]
+            BaseGroups_items([("myGroup", ("A", "B", "C")),
+            ("myGroup2", ("D", "E", "F")), ...])
+
         """
         return super(BaseGroups, self).items()
 
-    def keys(self):
-        """
-        Returns a ``list`` of all the group names in Groups. This list will be
-        unordered.::
+    def keys(self) -> BaseKeys:
+        """View the group names (keys) in the current groups instance.
+
+        :return: A :ref:`type-view` of :class:`str` items representing the groups' keys.
+
+        Example::
 
             >>> font.groups.keys()
-            ["myGroup4", "myGroup1", "myGroup5"]
+            BaseGroups_keys(["myGroup4", "myGroup1", "myGroup5", ...])
+
         """
         return super(BaseGroups, self).keys()
 
-    def pop(self, groupName, default=None):
+    def values(self) -> BaseValues:
+        """View the values in the current groups instance.
```

Comment:
As per other PR, maybe Return instead of View

Reply:
Fixed.